client/signup/steps/site-or-domain/index.jsx

```diff
@@ -18,21 +18,27 @@ import { getCurrentUserId } from 'state/current-user/selectors';
 import DomainImage from 'signup/steps/design-type-with-store/domain-image';
 import NewSiteImage from 'signup/steps/design-type-with-store/new-site-image';
 import ExistingSite from 'signup/steps/design-type-with-store/existing-site';
-import { externalRedirect } from 'lib/route/path';
 import NavigationLink from 'signup/navigation-link';
+import QueryProductsList from 'components/data/query-products-list';
+import { getProductsList } from 'state/products-list/selectors';
 
 class SiteOrDomain extends Component {
-	componentWillMount() {
-		if ( ! this.getDomainName() ) {
-			// /domains domain search is an external application to calypso,
-			// therefor a full redirect required:
-			externalRedirect( '/domains' );
+	getDomainProductSlug( domain ) {
+		const domainParts = domain.split( '.' );
+		const tld = domainParts.slice( 1 ).join( '.' );
+
+		let productSlug = null;
```

Comment:
You could just `return` ASAP and do away with `productSlug`:
```
if ( includes( [ 'com', 'net', 'org' ], tld ) ) {
	return 'domain_reg';
}

return `dot${ tld }_domain`;
```
(seems less complicated to me, but it's your call)